src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
```

Comment:
> We aren't sure this is the best API, and we want users to give us feedback after trying it

You're never going to be _sure_ of that. But I'm not aware of any major objections, or even alternative proposals, to the latest revision. (JUnit-Lambda doesn't count because it's vaporware.)

> Once we add an API to JUnit, it's hard to remove it, but it's possible we could have some Beta APIs JUnit-Lambda (possibly using an annotation like Guava's @Beta annotation). I think it's a good API, but it's hard to know what problems it has or what improvements to add until some people start using it.

You guys make it sound like you've _never added a feature to JUnit_ and you wouldn't even know where to start. This is ridiculous. I'm adding one (1) method to `org.junit.Assert`, following simple and well-established patterns that can probably be traced back to SUnit. The thing has a surface area the size of a postage stamp, and I'm already pretty confident that we've explored virtually the entire design space already.

Furthermore, you can't just worry about the risk of releasing `expectThrows` unless you're also going to consider the risk of _not_ releasing `expectThrows`. I've already pointed out that I've seen people writing these assertions by hand, over and over again, _in situ_ in a test method. The verbosity of having to do this is annoying, but the high odds of getting it wrong actually make it dangerous, too. I've seen people of all experience levels get this wrong to varying degrees. The longer you drag your feet, the more broken tests will be written.

> We are planning on building a lambda-based and/or fluent replacement for Assert/Assume/ErrorCollector in the near future

So what? That's no reason for development on the existing `Assert` class to come to a standstill. It might be a relevant objection to a competing proposal for some sort of massive overhaul, but that's not what this PR is about.